src/occupation.jl

```diff
@@ -89,11 +88,6 @@ function compute_occupation(basis::PlaneWaveBasis{T}, eigenvalues;
     end
 
     occupation = compute_occupation(basis, eigenvalues, εF)
-    minocc = maximum(minimum, occupation)
-    if temperature > 0 && minocc > occupation_threshold
-        @warn "One k-point has a high minimum occupation $minocc. You should probably increase the number of bands."
-    end
-
```

Comment:
that's great if it works, but it'd be also good to support the "old" way of doing things, without adaptativity

Reply:
this also really highlights the need to put "algorithmic details" into a struct

Reply:
> that's great if it works, but it'd be also good to support the "old" way of doing things, without adaptativity
 
I agree, but this is a strict improvement as you can still specify `n_bands` (and are guaranteed to get at least that many converged). The difference is just that now the code might decide to converge more automatically.

Reply:
It's still good to be able to specify a simple scheme (ie when the heuristics go crazy, or when you want to do perf benchmarking...)

Reply:
Ok, I changed that now. Turns out already on the magnesium example our default choice was flawed (some bands with rather reasonable occupation were not guaranteed to be fully converged).

Also I'll implement k-point specific `n_bands` in this PR, I think.

Reply:
> Also I'll implement k-point specific n_bands in this PR, I think.

Nah. That'll be a follow-up.